### ISSUE TITLE
How much memory is required?

### ISSUE THREAD

**Issue by yiyunchen** [yulunzhang/RNAN#4]
Thank you for your work. But when I try to run this code, it occur that "out of memory", I would like to know how much memory is required for this.

Thank you!

**Comment by yulunzhang** [comment]
Hi,

Different tasks/input sizes would need different GPU memory sizes. I don't remember exactly the GPU memory size of each task. But, all the experiments can be done with only one GPU ( 12 Gb memory).

You can first use a smaller batch size (like 4) or patch size (like 24) to make the code run through and make sure everything else is ok. Then, increase the batch size or patch size to be the same as the paper.

**Comment by circlehy** [comment]
> Hi,
> 
> Different tasks/input sizes would need different GPU memory sizes. I don't remember exactly the GPU memory size of each task. But, all the experiments can be done with only one GPU ( 12 Gb memory).
> 
> You can first use a smaller batch size (like 4) or patch size (like 24) to make the code run through and make sure everything else is ok. Then, increase the batch size or patch size to be the same as the paper.

I used the V100 GPU which has 32G memory, still out of memory training the DN-RGB model during validation. Any suggestion to fix the problem, smaller batch size it's not helping. Training takes no more than 10G GPU memory, while validation it is out of memory, I even change the validation set with smaller size images, still not helping.

**Comment by RichealYoung** [comment]
I solved this by add '--chop', which can crop the test image into size <=100*100, avoiding memory problem

**Comment by chenjiachengzzz** [comment]
> I solved this by add '--chop', which can crop the test image into size <=100*100, avoiding memory problem
i use --chop and 4 2080ti gpu but out of memory when validation. can you tell me your gpu seting, thanks

**Comment by circlehy** [comment]
Yeah, I found this '--chop' option and solved the problem at the time. Thanks.





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "yulunzhang/RNAN"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年9月18日(星期六) 晚上7:22
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [yulunzhang/RNAN] How much memory is required? (#4)





  
I solved this by add '--chop', which can crop the test image into size <=100*100, avoiding memory problem
 i use --chop and 4 2080ti gpu but out of memory when validation. can you tell me your gpu seting, thanks
  
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by chenjiachengzzz** [comment]
> Yeah, I found this '--chop' option and solved the problem at the time. Thanks.
> […](#)
> ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "yulunzhang/RNAN" ***@***.***&gt;; 发送时间:&nbsp;2021年9月18日(星期六) 晚上7:22 ***@***.***&gt;; ***@***.******@***.***&gt;; 主题:&nbsp;Re: [yulunzhang/RNAN] How much memory is required? (#4) I solved this by add '--chop', which can crop the test image into size <=100*100, avoiding memory problem i use --chop and 4 2080ti gpu but out of memory when validation. can you tell me your gpu seting, thanks — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.

i alread use --chop option and oom

**Comment by chenjiachengzzz** [comment]
> > Yeah, I found this '--chop' option and solved the problem at the time. Thanks.
> > […](#)
> > ------------------ 原始邮件 ------------------ 发件人: "yulunzhang/RNAN" _**@**_._**>; 发送时间: 2021年9月18日(星期六) 晚上7:22 _**@**_.**_>; _**@**_._**_**@**_.**_>; 主题: Re: [yulunzhang/RNAN] How much memory is required? (#4) I solved this by add '--chop', which can crop the test image into size <=100*100, avoiding memory problem i use --chop and 4 2080ti gpu but out of memory when validation. can you tell me your gpu seting, thanks — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.
> 
> i alread use --chop option and oom

CUDA_VISIBLE_DEVICES=4,5,6,7  python main.py --model RNAN --scale 2 --save RNAN_SR_F64G10P48BIX2 --save_results --chop --patch_size 96  --n_GPUs 4 --batch_size 32

**Comment by circlehy** [comment]
This issue was long time ago and I used one Tesla V100 to run the code. If it still out of memory, maybe you can crop the image into patches to go though the model. But I am not sure how they did it on&nbsp;Titan Xp GPU, as it mentioned in the paper.





------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "yulunzhang/RNAN"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年9月18日(星期六) 晚上8:07
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [yulunzhang/RNAN] How much memory is required? (#4)





   
Yeah, I found this '--chop' option and solved the problem at the time. Thanks.
 …
 ------------------&nbsp;原始邮件&nbsp;------------------ 发件人: "yulunzhang/RNAN" @.&gt;; 发送时间:&nbsp;2021年9月18日(星期六) 晚上7:22 _@**._&gt;; ***@***.***@.**_&gt;; 主题:&nbsp;Re: [yulunzhang/RNAN] How much memory is required? (#4) I solved this by add '--chop', which can crop the test image into size <=100*100, avoiding memory problem i use --chop and 4 2080ti gpu but out of memory when validation. can you tell me your gpu seting, thanks — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe. Triage notifications on the go with GitHub Mobile for iOS or Android.
  
i alread use --chop option and oom
  
CUDA_VISIBLE_DEVICES=4,5,6,7  python main.py --model RNAN --scale 2 --save RNAN_SR_F64G10P48BIX2 --save_results --chop --patch_size 96  --n_GPUs 4 --batch_size 32
 
—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
